extensions/wikipedia/src/search-page.tsx

```diff
@@ -21,16 +22,22 @@ export default function SearchPage() {
 }
 
 function PageItem({ title }: { title: string }) {
-  const { data: extract } = useWikipediaPageSummary(title);
+  const { data } = useWikipediaPageData(title);
+
   return (
     <List.Item
-      icon={Icon.TextDocument}
+      icon={{ source: data?.thumbnail?.source ? data?.thumbnail?.source : "../assets/wikipedia.png" }}
       id={title}
       key={title}
       title={title}
-      subtitle={extract}
+      subtitle={data?.description}
       actions={
         <ActionPanel>
+          <PushAction
+            icon={Icon.Window}
+            title={"Show Details"}
+            target={<ShowDetailsPage title={title} extract={data?.extract} description={data?.description} />}
+          />
```

Comment:
I think this should be either the secondary action or we should add a preference to toggle it.

Reply:
Works for me, I have moved `Show Details` action to secondary action.